Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -6398,6 +6399,150 @@ def return_warning(message, exit=False, warning='', outputs=None, ignore_auto_ex
     if exit:
         sys.exit(0)
 
+class ConcurrentCommandRunner:
+    """
+    Class for executing multiple commands and save the results of each command.
+    """
+    class CommandExecuter:
+        """
+        Class for the needed data to execute a command.
+        """
+        def __init__(self, commands, args_lst, brand=None, instance=None):
+            """
+
+            :param commands: The commands list or a single command
+            :type commands: List[str] + str
+            :param args_lst: The args list or a single args
+            :type args_lst: List[Dict] + Dict
+            :param brand: The brand to use
+            :type brand: str
+            :param instance: The instance to use
+            :type instance: str
+            """
+            if isinstance(commands, str) and isinstance(args_lst, dict):
+                commands = [commands]
+                args_lst = [args_lst]
+            if not isinstance(commands, list) or not isinstance(args_lst, list) or len(commands) != len(args_lst):
+                raise DemistoException('commands and arg_lst arguments given are not valid')
+            self.commands = commands
+            self.args_lst = args_lst
+            if brand:
+                for args in self.args_lst:
+                    args.update({'using-brand': brand})
+            if instance:
+                for args in self.args_lst:
+                    args.update({'using': instance})
+
+    class ResultWrapper:
+        """
+        Class for the result of the command.
```

Comment:
Since this is a data class, let's mention that.
Furthermore, since it's already a nested class, i.e. `ConcurrentCommandRunner.ResultWrapper` - let's remove the `wrapper` from the class name. In any case, I don't think `wrapper` is the right term for this entity.

Let's just call it `Result`.